pex/requirements.py

```diff
@@ -587,25 +519,29 @@ def _parse_requirement_line(
             specifier=specifier,
             marker=marker,
         )
-        return URLRequirement.create(line, url, requirement, editable=editable)
+        return URLRequirement(line, url, requirement, editable=editable)
 
     # Handle local archives and project directories via path or file URL (Pip proprietary).
     local_requirement = parsed_url._replace(scheme="").geturl()
     project_name_extras_and_marker = _try_parse_pip_local_formats(
         local_requirement, basepath=basepath
     )
     maybe_abs_path, extras, marker = (
-        project_name_extras_and_marker
+        (
+            project_name_extras_and_marker.project_name,
+            project_name_extras_and_marker.extras,
+            project_name_extras_and_marker.marker,
+        )
```

Comment:
```suggestion
        project_name_extras_and_marker.astuple()
```
It looks like we do this `xxx, extras, marker = (... if ... else ...)` deconstruction at least two times in this file, but I think that the `ProjectNameExtrasAndMarker.astuple()` method exists for exactly this reason.

Reply:
(this convo can be resolved at your discretion, but I'm pretty confident this change would make the result more readable. i would like to either use `.astuple()` here, or remove the awkward `if ... else ...` construct altogether)

Reply:
It does but it's only typed to Tuple[Any].

Reply:
Aha - I did not read this comment well. I glossed an assumed you meant the attrs API `astuple`: `attr.astuple(attrs_class_instance)`.  That is unfortunately typed to Tuple[Any] with the standard typeshed and that is where I started with all this. It took quite a bit of thrashing with mypy to learn this was the issue. That led to the explicit tuple style here then to writing my own astuple method that was properly typed, then reconsidering how smart that was but forgetting to delete my custom astuple.

Reply:
Fixed.